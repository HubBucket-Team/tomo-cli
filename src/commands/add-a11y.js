import {
    PackageJsonEditor,
    install,
    uninstall
} from '../utils';
import {allDoExist} from '../utils/common';
/**
 * @type {task[]}
 * @see http://pa11y.org/
 */
export const addA11y = [
    {
        text: 'Add pa11y tasks to package.json',
        task: async ({outputDirectory}) => {
            const scripts = {
                'lint:a11y': `pa11y ${outputDirectory}/index.html`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Install pa11y',
        task: ({skipInstall}) => install(['pa11y'], {dev: true, skipInstall}),
        condition: ({isNotOffline}) => isNotOffline && allDoExist('package.json')
    }
];
export const removeA11y = [
    {
        text: 'Remove pa11y tasks from package.json',
        task: async () => {
            const scripts = {
                'lint:a11y': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall pa11y',
        task: () => uninstall(['pa11y']),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll('pa11y'),
        optional: ({skipInstall}) => !skipInstall
    }
];
export default addA11y;