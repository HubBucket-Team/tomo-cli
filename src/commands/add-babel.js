import {
    allDoNotExist,
    BabelConfigModuleEditor,
    install,
    someDoExist
} from '../utils';

const cfg = new BabelConfigModuleEditor();

const BABEL_DEPENDENCIES = [
    '@babel/cli',
    '@babel/core',
    '@babel/runtime'
];
const BABEL_PRESETS = [
    '@babel/preset-env'
];
const BABEL_PLUGINS = [
    '@babel/plugin-transform-runtime',
    '@babel/plugin-proposal-class-properties',
    '@babel/plugin-proposal-export-default-from',
    '@babel/plugin-proposal-optional-chaining'
];
const BABEL_REACT_PRESET = [
    '@babel/preset-react'
];
export default [
    {
        text: 'Create Babel config file',
        task: () => cfg.create(),
        condition: () => allDoNotExist('babel.config.js', '.babelrc', '.babelrc.js')
    },
    {
        text: 'Install Babel dependencies',
        task: ({skipInstall}) => install(BABEL_DEPENDENCIES, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel presets',
        task: ({skipInstall}) => install(BABEL_PRESETS, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel plugins',
        task: ({skipInstall}) => install(BABEL_PLUGINS, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel React preset',
        task: ({skipInstall}) => install(BABEL_REACT_PRESET, {dev: true, skipInstall}),
        condition: ({useReact}) => (useReact && someDoExist('package.json')),
        optional: ({useReact}) => useReact
    },
    {
        text: 'Add React support to Babel configuration file',
        task: () => cfg.extend({
            presets: [...BABEL_PRESETS, ...BABEL_REACT_PRESET]
        }),
        condition: ({useReact}) => (useReact && someDoExist('babel.config.js')),
        optional: ({useReact}) => useReact
    }
];