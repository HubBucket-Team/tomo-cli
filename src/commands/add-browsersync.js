import {PackageJsonEditor, install} from '../utils';
import {allDoExist, someDoExist} from '../utils/common';

const BROWSERSYNC_DEPENDENCIES = [
    'browser-sync',
    'npm-run-all'
];
/**
 * @type {task[]}
 * @see https://www.browsersync.io/docs/command-line
 */
export const addBrowsersync = [
    {
        text: 'Add Browsersync tasks to package.json',
        task: async ({outputDirectory}) => {
            const scripts = {
                prestart: 'npm run build',
                start: 'npm-run-all --parallel build:watch build:css:watch serve:dev',
                'serve:dev': `browser-sync start --server ${outputDirectory} --files ${outputDirectory}`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json', 'webpack.config.js', 'postcss.config.js')
    },
    {
        text: 'Install Browsersync dependencies',
        task: ({skipInstall}) => install(BROWSERSYNC_DEPENDENCIES, {dev: true, skipInstall}),
        condition: ({isNotOffline}) => isNotOffline && someDoExist('package.json')
    }
];