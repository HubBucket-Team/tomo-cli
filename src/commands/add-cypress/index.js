import {join} from 'path';
import {EslintConfigModuleEditor, PackageJsonEditor, install, uninstall} from '../../utils';
import {allDoExist, allDoNotExist} from '../../utils/common';
import {Scaffolder} from '../../utils/Scaffolder';
import createJsonEditor from '../../utils/createJsonEditor';

const ALWAYS = () => true;
const CYPRESS_DEPENDENCIES = [
    'cypress',
    'cypress-match-screenshot'
];
/**
 * @type {task[]}
 * @see https://www.cypress.io/
 */
export const addCypress = [
    {
        text: 'Create Cypress config file',
        task: async ({port}) => {
            const Editor = createJsonEditor('cypress.json', {
                baseUrl: `http://localhost:${port}`
            });
            await (new Editor())
                .create()
                .commit();
        },
        condition: () => allDoNotExist('cypress.json')
    },
    {
        text: 'Add Cypress test tasks to package.json',
        task: async () => {
            const scripts = {
                cypress: 'cypress open',
                'test:e2e': 'npm-run-all --parallel start cypress'
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Add Cypress global variable to .eslintrc.js',
        task: async () => {
            const globals = {cy: true};
            await (new EslintConfigModuleEditor())
                .extend({globals})
                .commit();
        },
        condition: () => allDoExist('.eslintrc.js')
    },
    {
        text: 'Copy Cypress files',
        task: async ({overwrite}) => {
            await (new Scaffolder(join(__dirname, 'templates')))
                .overwrite(overwrite)
                .target('cypress/support')
                .copy('support/index.js', 'index.js')
                .copy('support/commands.js', 'commands.js')
                .target('cypress/integration')
                .copy('visual-regression.test.js')
                .commit();
        },
        condition: ALWAYS
    },
    {
        text: 'Install Cypress dependencies',
        task: ({skipInstall}) => install([...CYPRESS_DEPENDENCIES, 'npm-run-all'], {dev: true, skipInstall}),
        condition: ({isNotOffline}) => isNotOffline && allDoExist('package.json')
    }
];
export const removeCypress = [
    {
        text: 'Delete Cypress config file',
        task: async () => {
            const Editor = createJsonEditor('cypress.json');
            await (new Editor())
                .delete()
                .commit();
        },
        condition: () => allDoExist('cypress.json')
    },
    {
        text: 'Remove Cypress test tasks from package.json',
        task: async () => {
            const scripts = {
                cypress: undefined,
                'test:e2e': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Remove Cypress global variable from .eslintrc.js',
        task: async () => {
            const globals = {cy: false};
            await (new EslintConfigModuleEditor())
                .extend({globals})
                .commit();
        },
        condition: () => allDoExist('.eslintrc.js')
    },
    {
        text: 'Uninstall Cypress dependencies',
        task: () => uninstall(CYPRESS_DEPENDENCIES),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll(...CYPRESS_DEPENDENCIES),
        optional: ({skipInstall}) => !skipInstall
    }
];
export default addCypress;