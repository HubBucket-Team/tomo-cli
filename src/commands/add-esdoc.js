import {
    allDoNotExist,
    createJsonEditor,
    install,
    PackageJsonEditor,
    someDoExist
} from '../utils';

const ESDOC_CONF = {
    source: './src',
    destination: './docs',
    plugins: [
        {
            name: 'esdoc-standard-plugin'
        },
        {
            name: 'esdoc-ecmascript-proposal-plugin',
            option: {
                all: true
            }
        }
    ]
};
const ESDOC_DEPENDENCIES = [
    'esdoc',
    'esdoc-ecmascript-proposal-plugin',
    'esdoc-standard-plugin'
];
const ESDOC_REACT_PLUGINS = [
    'esdoc-jsx-plugin'
];
const EsdocJsonEditor = createJsonEditor('esdoc.conf.json', ESDOC_CONF);
const cfg = new EsdocJsonEditor();
const pkg = new PackageJsonEditor();
/**
 * @ignore
 */
export const tasks = [
    {
        text: 'Create esdoc configuration file',
        task: async () => {
            await cfg.create().commit();
        },
        condition: () => allDoNotExist('esdoc.conf.json', '.esdoc.json')
    },
    {
        text: 'Add documentation tasks to package.json',
        task: async ({sourceDirectory}) => {
            await pkg.extend({
                script: {
                    'lint:docs': `eslint . --no-eslintrc --rule valid-jsdoc:error --parser babel-eslint`,
                    'build:docs': `jsdoc ${sourceDirectory} -r --destination ./docs`,
                    'open:docs': 'opn ./docs/index.html',
                    predocs: 'npm run lint:docs',
                    docs: 'npm run build:docs',
                    postdocs: 'npm run open:docs'
                }
            }).commit();
        },
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install esdoc dependencies',
        task: ({skipInstall}) => install([...ESDOC_DEPENDENCIES, 'opn-cli'], {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install esdoc React plugins',
        task: ({skipInstall}) => install(ESDOC_REACT_PLUGINS, {dev: true, skipInstall}),
        condition: ({useReact}) => (useReact && someDoExist('package.json')),
        optional: ({useReact}) => useReact
    },
    {
        text: 'Add esdoc React plugin to configuration file',
        task: async () => {
            const {plugins} = ESDOC_CONF;
            await cfg
                .extend({
                    plugins: [...plugins, {
                        name: 'esdoc-jsx-plugin',
                        options: {enable: true}
                    }]
                })
                .commit();
        },
        condition: ({useReact}) => (useReact && someDoExist('esdoc.conf.json')),
        optional: ({useReact}) => useReact
    }
];
export default tasks;