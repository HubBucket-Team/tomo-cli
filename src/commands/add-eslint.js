import {
    allDoNotExist,
    install,
    EslintConfigModuleEditor,
    PackageJsonEditor,
    someDoExist
} from '../utils';

const pkg = new PackageJsonEditor();
const cfg = new EslintConfigModuleEditor();

const ESLINT_DEPENDENCIES = [
    'eslint',
    'babel-eslint',
    'eslint-config-omaha-prime-grade',
    'watch'
];
const ESLINT_REACT_PLUGINS = [
    'eslint-plugin-react'
];
export default [
    {
        text: 'Create ESLint config file',
        task: () => cfg.create(),
        condition: () => allDoNotExist('.eslintrc.js', '.eslintrc', '.eslintrc.json', '.eslintrc.yml')
    },
    {
        text: 'Add lint tasks to package.json',
        task: ({sourceDirectory}) => pkg.extend({
            script: {
                lint: `eslint -c ./.eslintrc.js ${sourceDirectory}/**/*.js --fix`,
                'lint:watch': `watch 'npm run lint' ${sourceDirectory}`
            }
        }),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install ESLint dependencies',
        task: ({skipInstall}) => install(ESLINT_DEPENDENCIES, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install ESLint React plugins',
        task: ({skipInstall}) => install(ESLINT_REACT_PLUGINS, {dev: true, skipInstall}),
        condition: ({useReact}) => (useReact && someDoExist('package.json')),
        optional: ({useReact}) => useReact
    },
    {
        text: 'Add React support to ESLint configuration file',
        task: ({reactVersion}) => cfg.extend({
            parserOptions: {
                ecmaFeatures: {
                    jsx: true
                }
            },
            settings: {
                react: {
                    version: `'${reactVersion}'`
                }
            },
            extends: ['omaha-prime-grade', 'plugin:react/recommended']
        }),
        condition: ({useReact}) => (useReact && someDoExist('.eslintrc.js')),
        optional: ({useReact}) => useReact
    }
];