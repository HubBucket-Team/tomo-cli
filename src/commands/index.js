/* eslint-disable no-magic-numbers */
import {
    createPackageJson,
    createSourceDirectory
} from './common';
import addA11y from './add-a11y';
import addBabel from './add-babel';
import addEsdoc from './add-esdoc';
import addEslint from './add-eslint';
import addJest from './add-jest';
import addMarionette from './add-marionette';
import addPostcss from './add-postcss';
import addWebpack from './add-webpack';

const createProject = [
    ...createPackageJson,
    ...createSourceDirectory,
    ...addBabel,
    ...addEslint,
    ...addJest
];
const create = {
    project: createProject,
    app: [
        ...createProject,
        ...addMarionette
    ],
    server: [
        ...createProject
    ]
};
const add = {
    a11y: addA11y,
    babel: addBabel,
    docs: addEsdoc,
    eslint: [
        ...addBabel,
        ...addEslint
    ],
    jest: [
        ...addBabel,
        ...addJest
    ],
    postcss: addPostcss,
    webpack: addWebpack
};

module.exports = {
    add,
    create,
    new: create// alias for create
};