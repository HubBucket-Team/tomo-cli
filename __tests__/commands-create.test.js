import {
    fileContents,
    getDirectoryTree,
    run,
    useTemporaryDirectory
} from './tomo-test';
import {
    createPackageJson,
    createSourceDirectory
} from '../src/commands/common';
import commands from '../src/commands';

jest.mock('is-online', () => (async () => true));

describe('"Create/New" commands', () => {
    let tempDirectory;
    const skipInstall = true;
    const {create} = commands;
    const [setTempDir, cleanupTempDir] = useTemporaryDirectory();
    beforeEach(async () => {
        tempDirectory = await setTempDir();
        process.chdir(tempDirectory);
    });
    afterEach(async () => {
        await cleanupTempDir();
    });
    test('create package.json', async () => {
        await run(createPackageJson, {});
        const contents = fileContents('./package.json');
        expect(contents).toMatchSnapshot();
    });
    test('create source directory', async () => {
        const sourceDirectory = 'some-random-folder-name';
        await run(createSourceDirectory, {sourceDirectory});
        const tree = getDirectoryTree(tempDirectory);
        expect(tree).toMatchSnapshot();
    });
    test('create new project', async () => {
        const sourceDirectory = 'src';
        const options = {skipInstall, sourceDirectory};
        await run(create.project, options);
        const tree = getDirectoryTree(tempDirectory);
        expect(tree).toMatchSnapshot();
    });
    test('create new app', async () => {
        const sourceDirectory = 'src';
        const outputDirectory = './dist';
        const options = {outputDirectory, skipInstall, sourceDirectory};
        await run(create.app, options);
        const tree = getDirectoryTree(tempDirectory);
        const pkg = fileContents('package.json');
        const cfg = fileContents('.eslintrc.js');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
        expect(cfg).toMatchSnapshot();
    });
    test('create new react app', async () => {
        const reactVersion = '16.2';
        const sourceDirectory = 'src';
        const outputDirectory = './dist';
        const useReact = true;
        const options = {outputDirectory, reactVersion, skipInstall, sourceDirectory, useReact};
        await run(create.app, options);
        const tree = getDirectoryTree(tempDirectory);
        const pkg = fileContents('package.json');
        const cfg = fileContents('.eslintrc.js');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
        expect(cfg).toMatchSnapshot();
    });
    test('create new server', async () => {
        const options = {skipInstall};
        await run(create.server, options);
        const tree = getDirectoryTree(tempDirectory);
        const pkg = fileContents('package.json');
        const cfg = fileContents('.eslintrc.js');
        expect(tree).toMatchSnapshot();
        expect(pkg).toMatchSnapshot();
        expect(cfg).toMatchSnapshot();
    });
});