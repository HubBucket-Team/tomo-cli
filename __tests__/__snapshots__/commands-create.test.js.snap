// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"Create/New" commands create new app 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 177,
            type: file
        },
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js,
                    size: 337,
                    type: file
                },
                {
                    name: setup.js,
                    size: 370,
                    type: file
                }
            ],
            size: 707,
            type: directory
        },
        {
            name: assets,
            children: [
                {
                    name: css,
                    children: [
                        {
                            name: style.css,
                            size: 238,
                            type: file
                        }
                    ],
                    size: 238,
                    type: directory
                },
                {
                    name: fonts,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                },
                {
                    name: images,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                },
                {
                    name: index.html,
                    size: 1131,
                    type: file
                },
                {
                    name: library,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                },
                {
                    name: workers,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                }
            ],
            size: 1369,
            type: directory
        },
        {
            name: babel.config.js,
            size: 322,
            type: file
        },
        {
            name: package.json,
            size: 1696,
            type: file
        },
        {
            name: postcss.config.js,
            size: 509,
            type: file
        },
        {
            name: src,
            children: [
                {
                    name: components,
                    children: [
                        {
                            name: app.js,
                            size: 1265,
                            type: file
                        }
                    ],
                    size: 1265,
                    type: directory
                },
                {
                    name: main.js,
                    size: 199,
                    type: file
                },
                {
                    name: plugins,
                    children: [
                        {
                            name: mn.radio.logging.js,
                            size: 3980,
                            type: file
                        },
                        {
                            name: mn.redux.state.js,
                            size: 2039,
                            type: file
                        }
                    ],
                    size: 6019,
                    type: directory
                },
                {
                    name: shims,
                    children: [
                        {
                            name: mn.renderer.shim.js,
                            size: 510,
                            type: file
                        }
                    ],
                    size: 510,
                    type: directory
                }
            ],
            size: 7993,
            type: directory
        },
        {
            name: webpack.config.js,
            size: 862,
            type: file
        }
    ],
    size: 13671,
    type: directory
}
"
`;

exports[`"Create/New" commands create new app 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"build\\": \\"webpack\\",
        \\"build:watch\\": \\"watch 'npm run build' src\\",
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --fix\\",
        \\"lint:watch\\": \\"watch 'npm run lint' src\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"build:css\\": \\"postcss ./assets/css/style.css --dir ./dist\\",
        \\"build:css:watch\\": \\"npm run build:css -- --watch\\",
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:watch\\": \\"npm test -- --watchAll\\",
        \\"copy\\": \\"npm-run-all --parallel copy:assets copy:index\\",
        \\"copy:assets\\": \\"cpy './assets/!(css)/**/*.*' './assets/**/[.]*' ./dist --parents --recursive\\",
        \\"copy:index\\": \\"cpy './assets/index.html' ./dist\\",
        \\"prebuild\\": \\"del-cli dist/assets\\",
        \\"postbuild\\": \\"npm run copy\\",
        \\"build:dashboard\\": \\"webpack-dashboard -- webpack-dev-server --config ./webpack.config.js\\",
        \\"dev\\": \\"stmux [ \\\\\\"npm run build:dashboard\\\\\\" : \\\\\\"npm run lint:watch\\\\\\" ]\\",
        \\"prestart\\": \\"npm run build\\",
        \\"start\\": \\"npm-run-all --parallel build:watch build:css:watch serve\\",
        \\"serve\\": \\"browser-sync start --server ./dist --files ./dist\\"
    },
    \\"jest\\": {
        \\"testMatch\\": [
            \\"**/__tests__/**/*.(e2e|test).[jt]s?(x)\\"
        ],
        \\"setupFilesAfterEnv\\": [
            \\"<rootDir>/__tests__/setup.js\\"
        ],
        \\"watchPlugins\\": [
            \\"jest-watch-typeahead/filename\\",
            \\"jest-watch-typeahead/testname\\"
        ]
    },
    \\"main\\": \\"src/main.js\\"
}
"
`;

exports[`"Create/New" commands create new app 3`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true,
        browser: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint'
};"
`;

exports[`"Create/New" commands create new project 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 154,
            type: file
        },
        {
            name: babel.config.js,
            size: 322,
            type: file
        },
        {
            name: package.json,
            size: 481,
            type: file
        },
        {
            name: src,
            children: [],
            size: 0,
            type: directory
        }
    ],
    size: 993,
    type: directory
}
"
`;

exports[`"Create/New" commands create new react app 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 375,
            type: file
        },
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js,
                    size: 337,
                    type: file
                },
                {
                    name: setup.js,
                    size: 370,
                    type: file
                }
            ],
            size: 707,
            type: directory
        },
        {
            name: assets,
            children: [
                {
                    name: css,
                    children: [
                        {
                            name: style.css,
                            size: 238,
                            type: file
                        }
                    ],
                    size: 238,
                    type: directory
                },
                {
                    name: fonts,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                },
                {
                    name: images,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                },
                {
                    name: index.html,
                    size: 1166,
                    type: file
                },
                {
                    name: library,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                },
                {
                    name: workers,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                }
            ],
            size: 1404,
            type: directory
        },
        {
            name: babel.config.js,
            size: 387,
            type: file
        },
        {
            name: package.json,
            size: 1648,
            type: file
        },
        {
            name: postcss.config.js,
            size: 509,
            type: file
        },
        {
            name: src,
            children: [
                {
                    name: components,
                    children: [
                        {
                            name: App.js,
                            size: 202,
                            type: file
                        }
                    ],
                    size: 202,
                    type: directory
                },
                {
                    name: main.js,
                    size: 220,
                    type: file
                }
            ],
            size: 422,
            type: directory
        },
        {
            name: webpack.config.js,
            size: 870,
            type: file
        }
    ],
    size: 6358,
    type: directory
}
"
`;

exports[`"Create/New" commands create new react app 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"build\\": \\"webpack\\",
        \\"build:watch\\": \\"watch 'npm run build' src\\",
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --fix\\",
        \\"lint:watch\\": \\"watch 'npm run lint' src\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"build:css\\": \\"postcss ./assets/css/style.css --dir ./dist\\",
        \\"build:css:watch\\": \\"npm run build:css -- --watch\\",
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:watch\\": \\"npm test -- --watchAll\\",
        \\"copy\\": \\"npm-run-all --parallel copy:assets copy:index\\",
        \\"copy:assets\\": \\"cpy './assets/!(css)/**/*.*' './assets/**/[.]*' ./dist --parents --recursive\\",
        \\"copy:index\\": \\"cpy './assets/index.html' ./dist\\",
        \\"prebuild\\": \\"del-cli dist/assets\\",
        \\"postbuild\\": \\"npm run copy\\",
        \\"build:dashboard\\": \\"webpack-dashboard -- webpack-dev-server --config ./webpack.config.js\\",
        \\"dev\\": \\"stmux [ \\\\\\"npm run build:dashboard\\\\\\" : \\\\\\"npm run lint:watch\\\\\\" ]\\",
        \\"serve\\": \\"webpack-dev-server --hot --open --mode development\\",
        \\"start\\": \\"npm-run-all --parallel build:css:watch serve\\"
    },
    \\"jest\\": {
        \\"testMatch\\": [
            \\"**/__tests__/**/*.(e2e|test).[jt]s?(x)\\"
        ],
        \\"setupFilesAfterEnv\\": [
            \\"<rootDir>/__tests__/setup.js\\"
        ],
        \\"watchPlugins\\": [
            \\"jest-watch-typeahead/filename\\",
            \\"jest-watch-typeahead/testname\\"
        ]
    },
    \\"main\\": \\"src/main.js\\"
}
"
`;

exports[`"Create/New" commands create new react app 3`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true,
        browser: true
    },
    extends: [
        'omaha-prime-grade',
        'plugin:react/recommended'
    ],
    parser: 'babel-eslint',
    parserOptions: {
        ecmaFeatures: {
            jsx: true
        }
    },
    settings: {
        react: {
            version: '16.2'
        }
    }
};"
`;

exports[`"Create/New" commands create new server 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .env,
            size: 0,
            type: file
        },
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 396,
            type: file
        },
        {
            name: .gitignore,
            size: 18,
            type: file
        },
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js,
                    size: 698,
                    type: file
                }
            ],
            size: 698,
            type: directory
        },
        {
            name: config,
            children: [
                {
                    name: default.js,
                    size: 857,
                    type: file
                },
                {
                    name: test.js,
                    size: 857,
                    type: file
                }
            ],
            size: 1714,
            type: directory
        },
        {
            name: favicon.ico,
            size: 0,
            type: file
        },
        {
            name: graphql.js,
            size: 1630,
            type: file
        },
        {
            name: index.js,
            size: 1434,
            type: file
        },
        {
            name: package.json,
            size: 996,
            type: file
        },
        {
            name: public,
            children: [
                {
                    name: example.md,
                    size: 263,
                    type: file
                },
                {
                    name: index.html,
                    size: 191,
                    type: file
                }
            ],
            size: 454,
            type: directory
        },
        {
            name: server.js,
            size: 2770,
            type: file
        },
        {
            name: socket.js,
            size: 624,
            type: file
        },
        {
            name: ssl,
            children: [
                {
                    name: server.cert,
                    size: 1241,
                    type: file
                },
                {
                    name: server.key,
                    size: 1704,
                    type: file
                }
            ],
            size: 2945,
            type: directory
        }
    ],
    size: 13715,
    type: directory
}
"
`;

exports[`"Create/New" commands create new server 2`] = `
"{
    \\"name\\": \\"tomo-server\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"Node.js HTTP(S), WebSocket, and GraphQL servers with an 80% solution for security 'baked in'\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"lint\\": \\"eslint . -c ./.eslintrc.js --fix\\",
        \\"lint:watch\\": \\"watch 'npm run lint' .\\",
        \\"lint:tests\\": \\"eslint __tests__/**/*.js -c ./.eslintrc.js --fix --no-ignore\\",
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:watch\\": \\"npm test -- --watchAll\\",
        \\"predev\\": \\"npm run open\\",
        \\"dev\\": \\"stmux [ \\\\\\"nodemon index.js\\\\\\" : \\\\\\"npm run lint:watch\\\\\\" ]\\",
        \\"prestart\\": \\"npm audit\\",
        \\"start\\": \\"node index.js\\",
        \\"open\\": \\"open-cli http://localhost:8111\\"
    },
    \\"jest\\": {
        \\"testMatch\\": [
            \\"**/__tests__/**/*.(e2e|test).[jt]s?(x)\\"
        ],
        \\"watchPlugins\\": [
            \\"jest-watch-typeahead/filename\\",
            \\"jest-watch-typeahead/testname\\"
        ]
    },
    \\"main\\": \\"index.js\\"
}
"
`;

exports[`"Create/New" commands create new server 3`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true,
        browser: false,
        node: true
    },
    extends: [
        'omaha-prime-grade',
        'plugin:react/recommended'
    ],
    parser: 'babel-eslint',
    parserOptions: {
        ecmaFeatures: {
            jsx: true
        }
    },
    settings: {
        react: {
            version: '16.2'
        }
    }
};"
`;

exports[`"Create/New" commands create package.json 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": []
}
"
`;

exports[`"Create/New" commands create source directory 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: some-random-folder-name,
            children: [],
            size: 0,
            type: directory
        }
    ],
    size: 0,
    type: directory
}
"
`;
